### PR TITLE
[Functions] Add deletion failure details

### PR DESCRIPTION
**Pull request by rokatyy**
Jira - https://iguazio.atlassian.net/browse/ML-6553

Add deletion failure details when function is being deleted via SDK

### CODE REVIEW

File: mlrun/db/httpdb.py [hunk 1]
```diff
@@ -1223,8 +1223,11 @@ def delete_function(self, name: str, project: str = ""):
                 background_task.status.state
                 == mlrun.common.schemas.BackgroundTaskState.failed
             ):
+                reason = background_task.status.error
                 logger.info(
-                    "Function deletion failed", project_name=project, function_name=name
+                    f"Function deletion failed. Reason: {reason}",
+                    project_name=project,
+                    function_name=name,
```
Comment (quaark) [inline mlrun/db/httpdb.py]:
```suggestion
                    f"Function deletion failed",
                    reason=background_task.status.error,
                    project_name=project,
                    function_name=name,
```

File: mlrun/db/httpdb.py [hunk 2]
```diff
@@ -1223,8 +1223,11 @@ def delete_function(self, name: str, project: str = ""):
                 background_task.status.state
                 == mlrun.common.schemas.BackgroundTaskState.failed
             ):
+                reason = background_task.status.error
                 logger.info(
-                    "Function deletion failed", project_name=project, function_name=name
+                    f"Function deletion failed. Reason: {reason}",
```
Comment (moranbental) [inline mlrun/db/httpdb.py]: I think it is better to include the `reason` as an argument, similar to `project_name`

**Review by moranbental — CHANGES_REQUESTED** [review]

**Review by quaark — COMMENTED** [review]
We usually put parameters in the logger parameters. Not a must if there's a specific reason to put it in the log message itself though

File: mlrun/db/httpdb.py (hunk 2 above)
Comment (rokatyy) [reply mlrun/db/httpdb.py]: @moranbental make sense! fixed

**Review by quaark — APPROVED** [review]
Great!

**Review by liranbg — APPROVED** [review]

**Review by moranbental — APPROVED** [review]
LGTM

**Pull request merged**